src/components/change-view/change-view.jsx

```diff
@@ -445,3 +434,20 @@ function loadDiffSettings () {
 function saveDiffSettings (settings) {
   layeredStorage.setItem(diffSettingsStorage, settings);
 }
+
+function findDiffType(from, to, page, stateDiffType = null) {
+  const preferredDiffType = stateDiffType || loadDiffType();
+  const relevantTypes = relevantDiffTypes(from, to, page);
+
+  return (relevantTypes.find(type => type.value === preferredDiffType))
+    ? preferredDiffType
+    : relevantTypes[0].value;
```

Comment:
No that wasn't intentional. I think I have it back to the intention of the original logic.
Basically, what I was doing was: 
```
pick the first defined of [stateDiffType, storedDiffType, defaultDiffType]
if that is a relevant diff type, return it
otherwise return the first relevant diff type
```
And what SHOULD happen is:
```
for each of [stateDiffType, storedDiffType, defaultDiffType] 
  if that is a relevant diff type, return it
if none of them were, return the first relevant diff type
```

correct?

Reply:
Yep! 👍